perspective.el

```diff
@@ -817,15 +793,18 @@ named collections of buffers and window configurations."
         (add-hook 'ido-make-buffer-list-hook 'persp-set-ido-buffers)
         (setq read-buffer-function 'persp-read-buffer)
         (mapc 'persp-init-frame (frame-list))
-        (setf (persp-buffers persp-curr) (buffer-list))
+        (setf (persp-buffers (frame-parameter nil 'persp-curr)) (buffer-list))
 
         (run-hooks 'persp-mode-hook))
     (ad-deactivate-regexp "^persp-.*")
     (remove-hook 'after-make-frame-functions 'persp-init-frame)
     (remove-hook 'ido-make-buffer-list-hook 'persp-set-ido-buffers)
     (setq read-buffer-function nil)
-    (setq perspectives-hash nil)
-    (setq global-mode-string (delq 'persp-modestring global-mode-string))))
+    (set-frame-parameter nil 'perspectives-hash nil)
+    (setq global-mode-string (delete '(:eval (persp-mode-line)) global-mode-string))
+    (set-default 'header-line-format (delete '(:eval (persp-mode-line)) header-line-format))
+    (unless (delete "" header-line-format)
+      (set-default 'header-line-format nil))))
```

Comment:
I don't understand what this statement is doing. Can you add a comment explaining?

Reply:
Added a comment to this line. It seems that setting `header-line-format` to an empty string still renders it in the buffer. Setting it to nil removes it.